### DISCUSSION TITLE
When executing the function "writeNormalEquation", "std:: bad_alloc" is displayed.

### DISCUSSION THREAD

**Discussion by IamLittleMonster** [groops-devs/groops#175] [Q&A] [answered]
Dear developer,

When I wanted to estimate the satellite ACO based on the example "scenario GnssNetwork", I made the following modifications to "030groovsGnssProcessing. xml" according to the relevant sections of the reference book:

1. Add satellite ACO as an unknown parameter:
![image](https://user-images.githubusercontent.com/51938228/228156455-8a5985e5-3016-4feb-a8e2-98e8f4c68432.png)

2. Set the satellite ACO to disable through selectParameters:
![image](https://user-images.githubusercontent.com/51938228/228156568-38bd0aaa-782a-4419-ba52-a0809efdefc2.png)

3.After estimating, set all parameters to enable for subsequent parameter estimation:
![image](https://user-images.githubusercontent.com/51938228/228156705-58f38284-4d2b-4e6c-980d-a7f404759952.png)

4.Write the normal equation:
![image](https://user-images.githubusercontent.com/51938228/228156874-dedf0e6b-eda5-4d09-811f-45e5d69fe49e.png)

However, when writing the normal equation, I encountered an ERROR of "std:: bad_alloc":
![image](https://user-images.githubusercontent.com/51938228/228157030-5e88d27d-7925-46e0-8b87-07a5beaf7bac.png)

I would like to ask if the developer is clear about how to resolve this issue, and whether this issue is related to the memory of my device or other reasons.

With Kind Regards,
Hongzhan

**Comment by zhedumi** [comment, marked as answer]
Hi,

issue is your memory. Check this thread https://github.com/groops-devs/groops/discussions/162 which has basically the same issue and how to resolve it.

best regards,
Patrick

**Reply by IamLittleMonster** [reply]
Dear Patrick,

Thank you so much!

According to your prompt, I have found a solution in #162 , but I have only tried to modify the relevant settings for selectorParameterizations after estimate:
![屏幕截图 2023-03-28 212150](https://user-images.githubusercontent.com/51938228/228252164-d2062a42-2bfd-46c9-867c-b357260c3bb3.png)
Next, I will try the third method you mentioned in #162 .

have a nice day,
Hongzhan
